### DISCUSSION TITLE
Nginx and Windows implementation

### DISCUSSION THREAD

**Discussion by Snarfty** [iam-TJ/open-dishy#3] [Q&A]
Hello,

Thanks for the fantastic work making this a reality.

I've been trying to get this to work with Nginx on Windows

I've found the reddit posts in regards to this and have managed to get it to the stage where it loads http://dishy.starlink.com/ from nginx but I just get this screen.

![image](https://github.com/user-attachments/assets/4ae6d124-6b24-4e35-8d6b-85582e78ce3f)

Very confident I just have a configs wrong but am a first time user of everything here.

I have this added to my nginx config

```
server {
 listen 80 default_server;
 listen [::]:80 default_server;
 server_name dishy.starlink.com;
 root C:/open-dishy-master/dishy.starlink.com/;
 index index.html;
}
server {
 listen 9201 default_server;
 listen [::]:9201 default_server;
 server_name dishy.starlink.com;
 root C:/open-dishy-master/dishy.starlink.com/;
 location / {
  proxy_pass http://192.168.100.1:9201;
  proxy_set_header Origin 'http://dishy.starlink.com';
  proxy_set_header Referrer 'http://dishy.starlink.com/';
 }
}
```

And C:\nginx\sites-available\dishy.starlink.com changed to this
```
# SPDX-License-Identifier: AGPL-3.0-or-later
server {
	listen 80;
	listen [::]:80;
	server_name dishy.starlink.com;
	root C:\open-dishy-master\dishy.starlink.com\;
	index C:\open-dishy-master\dishy.starlink.com\index.html;
  rewrite ^/statistics /index.html last;
	add_header Access-Control-Allow-Origin http://router.starlink.com:9001;
	add_header Vary Origin;
}
server {
	listen 9201;
	listen [::]:9201;
	server_name dishy.starlink.com;
	root C:\open-dishy-master\dishy.starlink.com\;
	location / {
		proxy_pass http://192.168.100.1:9201;
		proxy_set_header Origin 'http://dishy.starlink.com';
		proxy_set_header Referrer 'http://dishy.starlink.com/';
	}
}

```

Would anyone be able to give me some guidance on this?

**Comment by securitybryan** [comment]
I'll take a look a this if I get some time.  I am brand new to the architecture and how all this works.   I am thrilled to see a few people are as motivated as I am to get this working again.

**Comment by Snarfty** [comment]
It would appear that SpaceX has restored a level of functionality to the webgui.

![image](https://github.com/user-attachments/assets/65a19bfc-112e-4508-88ea-3c1f292e8d24)
